common/metrics/src/lib.rs

```diff
@@ -90,8 +104,44 @@ fn get_metrics_file<P: AsRef<Path>>(dir_path: &P, file_name: &str) -> File {
         .expect("Open metrics file failed")
 }
 
+pub fn gather_metrics() -> Vec<prometheus::proto::MetricFamily> {
+    let metric_families = prometheus::gather();
+    let mut kept_families: Vec<_> = Vec::new();
+    let mut total: usize = 0;
+    let mut kept_count: usize = 0;
+    for metric_family in metric_families {
+        let family_count = metric_family.get_metric().len();
+
```

Comment:
How about something like this so that we can identify the problematic families from the logs:

```
        if family_count > 100 {
            error!(
                "Metrics: Metric Family {} is trying to export too many metrics: {}. Please investigate this as one of the labels might have a very high cardinality.",
                metric_family.get_name(),
                family_count,
            );
        }

```

Reply:
I agree, this is useful, except that if we're using this metric and it goes from < 100 to > 100 it'll just disappear.